codegen/src/generator/query.rs

```diff
@@ -244,39 +244,103 @@ fn generate_struct_deserializer(name: &str, shape: &Shape) -> String {
             }}
         }}
 
-        try!(end_element ( tag_name , stack ));
+        try!(end_element(tag_name, stack));
 
         Ok(obj)
         ",
         name = name,
-        struct_field_deserializers = generate_struct_field_deserializers(shape),
+        struct_field_deserializers = generate_struct_field_deserializers(shape, service),
     )
 }
 
-fn generate_struct_field_deserializers(shape: &Shape) -> String {
+fn generate_struct_field_deserializers(shape: &Shape, service: &Service) -> String {
+    let mut flattened_stuffs = match shape.member {
+        Some(ref flattened_member) => format!("flattened member: {:?}", flattened_member),
+        None => "nada".to_string(),
+    };
+
     shape.members.as_ref().unwrap().iter().map(|(member_name, member)| {
-        format!(
-            "\"{member_name}\" => {{
-                obj.{field_name} = {parse_expression};
-                continue;
-            }}",
-            field_name = member_name.to_snake_case(),
-            parse_expression = generate_struct_field_parse_expression(shape, member_name, member),
-            member_name = member_name,
-        )
+        // debug garbage:
+        // look up member.shape in all_shapes.  use that shape.member.location_name
+        let mut location_name = member_name.to_string();
+        let mut use_loc_name = false;
+
+        match service.shape_for_member(member) {
```

Comment:
This is the worst of things I did to get code working.  The idea was to check if the member we're dealing with has a child shape that is flattened.  If it is, look up that child member and if it has a location_name specified, use that.

Much cleanup can be done by using .some() and if statements instead of nested matchers with empty arms.